spec/support/tempfile.cr

```diff
@@ -44,14 +47,24 @@ def with_temp_executable(name, file = __FILE__)
 end
 
 def with_temp_c_object_file(c_code, *, filename = "temp_c", file = __FILE__)
-  obj_ext = {{ flag?(:win32) ? ".obj" : ".o" }}
+  obj_ext = {{ flag?(:msvc) ? ".obj" : ".o" }}
   with_tempfile("#{filename}.c", "#{filename}#{obj_ext}", file: file) do |c_filename, o_filename|
     File.write(c_filename, c_code)
 
-    {% if flag?(:win32) %}
-      `cl.exe /nologo /c #{Process.quote(c_filename)} #{Process.quote("/Fo#{o_filename}")}`.should be_truthy
+    {% if flag?(:msvc) %}
+      # following is based on `Crystal::Compiler#linker_command`
+      cl = "cl.exe"
+
+      if msvc_path = Crystal::System::VisualStudio.find_latest_msvc_path
+        # we won't be cross-compiling the specs binaries, so host and target
+        # bits are identical
+        bits = {{ flag?(:bits64) ? "x64" : "x86" }}
+        cl = Process.quote(msvc_path.join("bin", "Host#{bits}", bits, "cl.exe").to_s)
+      end
+
+      `#{cl} /nologo /c #{Process.quote(c_filename)} #{Process.quote("/Fo#{o_filename}")}`.should be_truthy
     {% else %}
-      `#{ENV["CC"]? || "cc"} #{Process.quote(c_filename)} -c -o #{Process.quote(o_filename)}`.should be_truthy
+      `#{Process.quote(ENV["CC"]? || "cc")} #{Process.quote(c_filename)} -c -o #{Process.quote(o_filename)}`.should be_truthy
```

Comment:
I think it's invalid to quote the CC variable, I'm pretty sure I've seen it being set to a compiler **with a set of flags**